lib/matplotlib/axes/_secondary_axes.py

```diff
@@ -43,11 +43,10 @@ def _make_secondary_locator(rect, parent):
     *parent*.
     """
     _rect = mtransforms.Bbox.from_bounds(*rect)
-    bb = mtransforms.TransformedBbox(_rect, parent.transAxes)
-    tr = parent.figure.transFigure.inverted()
-    bb = mtransforms.TransformedBbox(bb, tr)
-
     def secondary_locator(ax, renderer):
+        bb = mtransforms.TransformedBbox(_rect, parent.transAxes)
```

Comment:
@anntzer I *knew* there was a reason for this song and dance...

Reply:
Maybe add a comment mentioning this?

Reply:
Good idea. Comment added.